app/src/main/java/org/hackillinois/android/view/OutdoorMapsFragment.java

```diff
@@ -19,70 +24,133 @@
+        getUserLocation();
 
-    private boolean mapsAvailable() {
-        return GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(getActivity()) == ConnectionResult.SUCCESS;
+        return view;
     }
 
     public void onMapReady(GoogleMap map) {
         this.map = map;
         map.getUiSettings().setIndoorLevelPickerEnabled(false);
+        map.addMarker(new MarkerOptions().position(BUILDING_LOCATIONS[index]));
+        map.moveCamera(CameraUpdateFactory.newLatLngZoom(BUILDING_LOCATIONS[index], ZOOM));
+    }
 
-        map.addMarker(new MarkerOptions().position(BUILDING_LOCATIONS[0]));
-        map.moveCamera(CameraUpdateFactory.newLatLngZoom(BUILDING_LOCATIONS[0], ZOOM));
-        building.setText(BUILDING_NAMES[0]);
+    private void getUserLocation() {
+        if (ContextCompat.checkSelfPermission(getContext(), Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
+            final int ONE_MINUTE = 60 * 1000;
+            userLocation = new UserLocation();
+            locationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER, ONE_MINUTE, 0, userLocation);
+            locationInfo.setVisibility(View.VISIBLE);
+        } else {
+            requestPermissions(new String[]{Manifest.permission.ACCESS_FINE_LOCATION}, LOCATION_REQUEST_CODE);
+        }
+    }
+
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        if (requestCode == LOCATION_REQUEST_CODE && permissions[0].equals(Manifest.permission.ACCESS_FINE_LOCATION) && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+            getUserLocation();
+            locationInfo.setVisibility(View.VISIBLE);
+        }
     }
 
     private class TabListener implements TabLayout.OnTabSelectedListener {
         public void onTabSelected(TabLayout.Tab tab) {
-            int index = tab.getPosition();
+            index = tab.getPosition();
             map.addMarker(new MarkerOptions().position(BUILDING_LOCATIONS[index]));
             map.animateCamera(CameraUpdateFactory.newLatLngZoom(BUILDING_LOCATIONS[index], ZOOM));
+
             building.setText(BUILDING_NAMES[index]);
+            if (userLocation != null) {
+                userLocation.update();
+            }
         }
 
         public void onTabUnselected(TabLayout.Tab tab) {
             map.clear();
         }
 
         public void onTabReselected(TabLayout.Tab tab) {
-            int index = tab.getPosition();
+            index = tab.getPosition();
             map.animateCamera(CameraUpdateFactory.newLatLngZoom(BUILDING_LOCATIONS[index], ZOOM));
         }
     }
+
+    private class UserLocation implements android.location.LocationListener {
+        private LatLng location;
+
+        public void onLocationChanged(Location location) {
+            this.location = new LatLng(location.getLatitude(), location.getLongitude());
+            update();
+        }
+
+        private void update() {
```

Comment:
GoogleMapsUtil provides a utility function, SphericalUtil.computeDistanceBetween(), which we could also use to be consistent with google maps. http://googlemaps.github.io/android-maps-utils/javadoc/

Probably not worth adding another dependency for, but up to you.

Reply:
Right now I'm using the manhattan distance between the two points, since I figured that's a good estimate of the distance someone would take while walking. I might use what you suggested (or maybe the distance matrix api) but first I'm gonna try it out when I get back on campus.